src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_garr.cpp

```diff
@@ -11,76 +11,155 @@ SDComment: Adds NYI
 SDCategory: Molten Core
 EndScriptData */
 
+#include "Containers.h"
 #include "molten_core.h"
-#include "ObjectMgr.h"
+#include "ObjectAccessor.h"
 #include "ScriptedCreature.h"
 #include "ScriptMgr.h"
+#include "SpellInfo.h"
+
+enum Texts
+{
+    EMOTE_MASS_ERRUPTION        = 0,
+};
 
 enum Spells
 {
     // Garr
-    SPELL_ANTIMAGIC_PULSE   = 19492,
-    SPELL_MAGMA_SHACKLES    = 19496,
-    SPELL_ENRAGE            = 19516,
+    SPELL_ANTIMAGIC_PULSE       = 19492,
+    SPELL_MAGMA_SHACKLES        = 19496,
+    SPELL_ENRAGE                = 19516,
 
-    // Adds
-    SPELL_ERUPTION          = 19497,
-    SPELL_IMMOLATE          = 20294,
+    // Fireworn
+    SPELL_SEPARATION_ANXIETY    = 23492,
+    SPELL_ERUPTION              = 19497,
+    SPELL_MASSIVE_ERUPTION      = 20483,
+    SPELL_ERUPTION_TRIGGER      = 20482,    // Removes banish auras and applied immunity to banish
 };
 
 enum Events
 {
     EVENT_ANTIMAGIC_PULSE    = 1,
-    EVENT_MAGMA_SHACKLES     = 2,
+    EVENT_MAGMA_SHACKLES,
+};
+
+enum Creatures
+{
+    NPC_FIRESWORN            = 12099,
 };
 
 class boss_garr : public CreatureScript
 {
 public:
-    boss_garr() : CreatureScript("boss_garr") { }
+    boss_garr() : CreatureScript("boss_garr")
+    {
+    }
 
     struct boss_garrAI : public BossAI
     {
-        boss_garrAI(Creature* creature) : BossAI(creature, BOSS_GARR)
+        boss_garrAI(Creature* creature) : BossAI(creature, BOSS_GARR),
+                                          massEruptionTimer(600000)
+        {
+        }
+
+        void Reset() override
+        {
+            _Reset();
+            massEruptionTimer = 600000;
+
+            std::list<Creature*> fireworns;
+            me->GetCreatureListWithEntryInGrid(fireworns, NPC_FIRESWORN, 240.0f);
+            if (!fireworns.empty())
+            {
+                for (Creature* fireworn : fireworns)
```

Comment:
feel free to use `auto` in cases like this
```suggestion
                for (auto fireworn : fireworns)
```